packages/airnode-deployer/src/infrastructure/index.ts

```diff
@@ -260,91 +290,208 @@ function transformTerraformOutput(terraformOutput: string): DeployAirnodeOutput
   );
 }
 
-async function deploy({
-  airnodeAddressShort,
-  stage,
-  cloudProvider,
-  httpGateway,
-  httpSignedDataGateway,
-  configPath,
-  secretsPath,
-  airnodeWalletPrivateKey,
-}: AirnodeDeployParams): Promise<DeployAirnodeOutput> {
+export const deployAirnode = async (config: Config, configPath: string, secretsPath: string) => {
+  const { airnodeWalletMnemonic, cloudProvider, stage } = config.nodeSettings;
+  const airnodeAddress = deriveAirnodeAddress(airnodeWalletMnemonic);
+  const { type, region } = cloudProvider as CloudProvider;
+
+  const spinner = logger.getSpinner().start(`Deploying Airnode ${airnodeAddress} ${stage} to ${type} ${region}`);
   if (logger.inDebugMode()) {
     spinner.info();
   }
 
-  const { type: cloudProviderType } = cloudProvider;
-  const bucket = getBucketName(airnodeAddressShort, stage);
-  const terraformStateCloudProviderDir = path.join(terraformStateDir, cloudProviderType);
-
-  if (!(await cloudProviderLib[cloudProviderType].stateExists(bucket, cloudProvider as any))) {
-    // Run state recipes
-    logger.debug('Running state Terraform recipes');
-    const stateTmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'airnode'));
-
-    const execOptions = { cwd: stateTmpDir };
-    await execTerraform(execOptions, 'init', [['from-module', terraformStateCloudProviderDir]]);
-
-    const commonArguments: CommandArg[] = [
-      ['var', 'airnode_address_short', airnodeAddressShort],
-      ['var', 'stage', stage],
-      ['input', 'false'],
-      'no-color',
-      'auto-approve',
-    ];
-    await execTerraform(execOptions, 'apply', prepareAirnodeManageArguments(cloudProvider, commonArguments));
+  const goDeploy = await go(async () => {
+    logger.debug('Fetching Airnode bucket');
+    let bucketName = await cloudProviderLib[type].getAirnodeBucket(cloudProvider as any);
```

Comment:
I'm just checking; a mutable with an if and set is preferred over a "IIFE"?
I don't mind relaxed mutability but we should probably make it a policy or something.